addons/mass_mailing/models/ir_model.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class IrModel(models.Model):
+    _inherit = 'ir.model'
+
+    is_mailing = fields.Boolean(
+        string="Mailing Enabled",
+        compute='_compute_is_mailing', search='_search_is_mailing',
+        help="Whether this model supports marketing mailing capabilities (notably email and SMS).",
+    )
+
+    def _compute_is_mailing(self):
+        for model in self:
+            model.is_mailing = getattr(self.env[model.model], '_mailing_enabled', False)
+
+    def _search_is_mailing(self, operator, value):
+        valid_models = self.env['ir.model']
+        for model in self.search([]):
+            if model.model not in self.env or model.is_transient():
+                continue
+            if getattr(self.env[model.model], '_mailing_enabled', False):
+                valid_models |= model
+
+        search_positive = (operator == '=' and value) or (operator == '!=' and not value)
```

Comment:
Aren't you suppose to throw some kind of unsupported exception when operator is not in ('=', '!=') ?

Side note: "search_positive" is meh, "search_is_mailing" ?

Reply:
Well it is a boolean, not sure we could receive something else ... but I added a ValueError just in case. Updated search_positive.